### ISSUE TITLE
[custom emoticons] allow edit/delete from the menu

### ISSUE THREAD

**Issue by c10ud** [emesene/emesene#95]
gtk

**Comment by c10ud** [comment]
```
def event(self, button, event, smilie):
    if event.type == gtk.gdk.BUTTON_PRESS and event.button == 3:

        # -- Custom Emoticon Menu --
        self.emoMenu = gtk.Menu()
        self.emoMenu.connect('enter-notify-event', self.on_enter_notify_event)

        self.shortcutItem = gtk.ImageMenuItem(_("Change shortcut"))
        self.shortcutItem.connect('enter-notify-event', self.on_enter_notify_event)
        self.shortcutItem.set_image(gtk.image_new_from_stock(
            gtk.STOCK_EDIT, gtk.ICON_SIZE_MENU))
        self.shortcutItem.connect("activate", self.onEditShortcut,
            smilie)

        self.deleteItem = gtk.ImageMenuItem(_("Delete"))
        self.deleteItem.connect('enter-notify-event', self.on_enter_notify_event)
        self.deleteItem.set_image(gtk.image_new_from_stock(
            gtk.STOCK_DELETE, gtk.ICON_SIZE_MENU))
        self.deleteItem.connect("activate", self.onDeleteEmo, smilie)

        self.emoMenu.add(self.shortcutItem)
        self.emoMenu.add(self.deleteItem)

        self.emoMenu.show_all()
        # -----

        self.emoMenu.popup(None, None, None, event.button, event.time)
```

**Comment by c10ud** [comment]
just find out where to hook the right click signal with custom emoticons gtk buttons and adapt that emesene-1 code for the good

**Comment by c10ud** [comment]
Custom emoticons: allow edit/delete from contextual menu. Closed by 4851ef2aa6fa4d1112ab478ae0baa94530bda04f

Exposed two handy methods from EmoticonsCache for this kind of manipulation
Added a nice gtk.Menu for editing/deleting
